test/modules/TestResultCoordinator/Reports/TestReportGeneratorFactory.cs

```diff
@@ -183,11 +184,18 @@ internal TestReportGeneratorFactory(ITestOperationResultStorage storage, Network
             }
         }
 
-        async Task<NetworkStatusTimeline> GetNetworkStatusTimelineAsync(TimeSpan tolerancePeriod)
+        async Task<Option<NetworkStatusTimeline>> GetNetworkStatusTimelineAsync(TimeSpan tolerancePeriod)
         {
-            return await NetworkStatusTimeline.CreateAsync(
-                new StoreTestResultCollection<TestOperationResult>(this.Storage.GetStoreFromSource("networkController"), BatchSize),
-                tolerancePeriod);
+            try
+            {
+                return Option.Some(await NetworkStatusTimeline.CreateAsync(
+                    new StoreTestResultCollection<TestOperationResult>(this.Storage.GetStoreFromSource("networkController"), BatchSize),
+                    tolerancePeriod));
+            }
+            catch (InvalidDataException)
+            {
+                return Option.None<NetworkStatusTimeline>();
+            }
```

Comment:
Letting this fail and then returning None seems like it could lead to some unintended side effects. What if we give the TRC some configuration that could tell it that it shouldn't be dealing with this network status timeline?